### ISSUE TITLE
create a new texture manager api

### ISSUE THREAD

**Issue by joshmarinacci** [joshmarinacci/voxeljs-next#26]
currently it's not clear how textures are loaded and how they are mapped onto blocks. we need a new api.

**Comment by joshmarinacci** [comment]
this new api should let you

* specify the full url to an image, possibly even remote
* indicate if a texture is animated and how many frames
* for block types with multiple sides that are different (ex. grass on top and grass mixed w/ dirt for the sides), specify each of those sides
* indicate if the image has transparent parts (cut-outs, not true alpha)

**Comment by drBenway** [comment]
Hi Josh,
I was looking at your code and noticed this issue.
Would it be beneficial for speed if textures are grouped in sprite sheets and injected in the code(build) via base64 encoding? If you then serve your code via gzipped file I guess the download would be faster.

**Comment by joshmarinacci** [comment]
Yep. It definitely should support loading tile textures from a spritesheet. I don't know if base64 encoding would help, though. PNG is pretty compact already.
